src/MSBuild/XMake.cs

```diff
@@ -2586,6 +2586,14 @@ static bool DoesEnvironmentSupportTerminalLogger()
                         new BuildManager.DeferredBuildMessage("TerminalLogger was not used because the output is not supported.", MessageImportance.Low));
                     return false;
                 }
+
+                if (Traits.Instance.EscapeHatches.EnsureStdOutForChildNodesIsPrimaryStdout)
+                {
+                    s_globalMessagesToLogInBuildLoggers.Add(
+                        new BuildManager.DeferredBuildMessage("TerminalLogger was not used because build is run in context of process (e.g. 'dotnet test') that requests direct access to stdout stream.", MessageImportance.Low));
```

Comment:
We should have an issue for localizing these strings.

Reply:
I was just blindly following suit of this method :-)
Let me add all the messages from this method to resources as part of this PR

Reply:
Localized: https://github.com/dotnet/msbuild/pull/9066/commits/0b789d2798d2e0ca267e234a5e4c2643ccbe3a7d

Reply:
Thank you for doing this!